examples/regression/4-tutorials/plot_cqr_tutorial.py

```diff
@@ -0,0 +1,446 @@
+
+
+df = pd.concat([X, y], axis=1)
+pear_corr = df.corr(method='pearson')
+pear_corr.style.background_gradient(cmap='Greens', axis=0)
+
+
+##############################################################################
+# Now let's visualize a histogram of the price of the houses.
+
+
+fig, axs = plt.subplots(1, 1, figsize=(5, 5))
+axs.hist(y, bins=50)
+axs.set_xlabel("Median price of houses")
+axs.set_title("Histogram of house prices")
+axs.xaxis.set_major_formatter(FormatStrFormatter('%.0f'+"k"))
+plt.show()
+
+
+##############################################################################
+# Let's now create the different splits for the dataset, with a training,
+# calibration and test set. Recall that the calibration set is used for
+# calibrating the prediction intervals.
+
+
+X_train, X_test, y_train, y_test = train_test_split(
+    X,
+    y['MedHouseVal'],
+    random_state=random_state
+)
+X_train, X_calib, y_train, y_calib = train_test_split(
+    X_train,
+    y_train,
+    random_state=random_state
+)
+
+
+##############################################################################
+# 2. Optimizing estimator
+# --------------------------------------------------------------------------
+# Before estimating uncertainties, let's start by optimizing the base model
+# in order to reduce our prediction error. We will use the
+# :class:`~lightgbm.LGBMRegressor` in the quantile setting. The optimization
+# is performed using :class:`~sklearn.model_selection.RandomizedSearchCV`
+# to find the optimal model to predict the house prices.
+
+
+estimator = LGBMRegressor(
+    objective='quantile',
+    alpha=0.5,
+    random_state=random_state
+)
+params_distributions = dict(
+    num_leaves=randint(low=10, high=50),
+    max_depth=randint(low=3, high=20),
+    n_estimators=randint(low=50, high=300),
+    learning_rate=uniform()
+)
+optim_model = RandomizedSearchCV(
+    estimator,
+    param_distributions=params_distributions,
+    n_jobs=-1,
+    n_iter=100,
+    cv=KFold(n_splits=5, shuffle=True),
+    verbose=-1
+)
+optim_model.fit(X_train, y_train)
+estimator = optim_model.best_estimator_
+
+
+##############################################################################
+# 3. Comparison of MAPIE methods
+# --------------------------------------------------------------------------
+# We will now proceed to compare the different methods available in MAPIE used
+# for uncertainty quantification on regression settings. For this tutorial we
+# will compare the "naive", "Jackknife plus after Bootstrap", "cv plus" and
+# "conformalized quantile regression". Please have a look at the theoretical
+# description of the documentation for more details on these methods.
+#
+# We also create two functions, one to sort the dataset in increasing values
+# of ``y_test`` and a plotting function, so that we can plot all predictions
+# and prediction intervals for different conformal methods.
+
+
+def sort_time_series(y_test, y_pred, y_pis):
```

Comment:
Unless I'm mistaken, you are not using a Time Series dataset, so the name of the function is wrong

Reply:
True, I will change it.